### PR TITLE
compute: move logic to enterprise directory

### PR DESCRIPTION
**Pull request by rvantonder**
I started prototyping `compute` and logic outside the enterprise directories. It should probably live as part of the enterprise layer unless we decide otherwise.

I tested locally that the endpoint works against enterprise, etc. @coury-clark any chance you can provide some step-by-step procedure to just test this works against code insights still?

Details:

- This moves `internal/compute` => `enterprise/internal/compute` (main logic)
- Adds `enterprise/cmd/frontend/internal/compute/resolvers/resolvers.go` in enterprise that NO JOKE took me like 45 minutes just to update the types so I can references the abstracted `interface` types in `graphqlbackend` packages and keep the private resolvers by obnoxiously balancing interface/pointers. i dislike all of this fragmented referencing.
- Hooks compute schema into the enterprise checking code

If you're aware of any "gotchas" as far as enterprise and GQL resolvers go, please have a look out for those in this PR. Like I said I tested things and I think I followed all the boiler conventions we do elsewhere, but it was pretty fatiguing.

### CODE REVIEW

**Comment by rvantonder** [comment]
Adding @benvenker for visibility to this change

**Comment by camdencheek** [comment]
Oof yeah. Hooking in the enterprise code is kinda a pain. Will take a look 👍

**Comment by rvantonder** [comment]
To elaborate on manual tests: the endpoint works with `sg start enterprise` but does not work (as desired) for `sg start oss`.

**Comment by rvantonder** [comment]
@camdencheek no rush I'm signing off for the night :-)

File: enterprise/cmd/frontend/internal/compute/resolvers/resolvers.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+func (c *computeTextResolver) Repository() *gql.RepositoryResolver { return c.repository }
+
+func (c *computeTextResolver) Commit() *string {
+	value := c.commit
+	return &value
+}
+
+func (c *computeTextResolver) Path() *string {
+	value := c.path
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/compute/resolvers/resolvers.go:111]: Why the copies here?

File: enterprise/cmd/frontend/internal/compute/resolvers/resolvers.go [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+	return res, ok
+}
+
+func (r *computeResultResolver) ToComputeText() (gql.ComputeTextResolver, bool) {
+	res, ok := r.result.(*computeTextResolver)
+	return res, ok
+}
+
+func toComputeMatchContextResolver(mc *compute.MatchContext, repository *gql.RepositoryResolver, path, commit string) *computeMatchContextResolver {
+	var computeMatches []gql.ComputeMatchResolver
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/compute/resolvers/resolvers.go:141]: 🎵 _preallocating is super cooooool_ 🎵

File: enterprise/cmd/frontend/internal/compute/resolvers/resolvers_test.go [hunk 3]
```diff
@@ -30,8 +30,11 @@ func TestToResultResolverList(t *testing.T) {
 		)
 		var results []string
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/compute/resolvers/resolvers_test.go:31]: 🎵 _super awesome is preallocating_ 🎵

**Review by camdencheek — APPROVED** [review]
I just realized the comments I left aren't actually on things that were added by this PR, so feel free to ignore them and merge this. 

The enterprise hooks look good to me, and you tested them and they work, most things will loudly fail if you do it wrong (schema validation works quite well)

**Review by novoselrok — APPROVED** [review]
I definitely feel the pain of moving stuff to enterprise 😄

**Comment by coury-clark** [comment]
@rvantonder Code Insights uses the GraphQL API - so if the API works in the enterprise configuration we should be set. I poked around on the branch and it seems to be working:
image

To run this you just have to run a slightly different `sg` command `sg start enterprise-codeinsights`, then navigate to the `insights` tab and create a `Detect and track` insight type.

**Review by coury-clark — APPROVED** [review]

**Comment by rvantonder** [comment]
Thank you all! ❤️

**Pull request merged**